packages/manager/apps/web/client/app/hosting/offer/upgrade/hosting-offer-upgrade.controller.js

```diff
@@ -110,40 +110,55 @@ angular.module('App').controller(
       );
     }
 
-    getDetachActionsOptions() {
-      return {
-        addons: [
-          {
+    getDetachRequestOptions(serviceId, planCode) {
+      return this.ovhManagerProductOffersActionService
+        .getDetachPlancodeInformationOptions(serviceId, planCode)
+        .then((data) => {
+          const addons = [];
+          if (data.length === 1) {
```

Comment:
addons is plural but we can't have more than 2 or at least we don't handle the case ? Maybe we should `map` data no ?

Reply:
I agree this is weird, but it comes from the api design. 
We can only have 0 or one addon to be handled in detach.
But the detach apis do not reflect this design (paramater "addons" is an array, not an object)

Reply:
yes, you are right. even if for the moment we only either 0 or one addon, we can easily be more flexible